plugins/recommendations-plugin/src/plugin/recommendations-plugin.ts

```diff
@@ -126,9 +126,15 @@ export class RecommendationsPlugin {
   // called after projects are cloned (like the first import)
   async afterClone(): Promise<void> {
     // current workspaces
-    const workspaceFolders = theia.workspace.workspaceFolders || [];
+    let workspaceFolders = theia.workspace.workspaceFolders || [];
+    // see https://github.com/eclipse/che/issues/19624
+    if (workspaceFolders.length === 0) {
+      console.log('Delay workspace folders resolution as there is no workspace folder');
+      await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
WDYT about waiting for workspace plugin to being initialized instead of adding this timeout?

Reply:
@vitaliy-guliy how de we know when workspace plug-in is initialized ? (also initialized is after clone has happened ? )

Reply:
I suppose workspace plugin should be completely initialized only after cloning all the projects.
And if we add a plugin dependency to `package.json`, the recommendation plugin will wait for workspace plugin.

Reply:
there is already a dependency to the plug-in in package.json